src/command_modules/azure-cli-datalake/azure/cli/command_modules/datalake/store/_params.py

```diff
@@ -0,0 +1,53 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from azure.cli.core.commands import \
+    (register_cli_argument, CliArgumentType)
+
+from azure.cli.core.commands.parameters import \
+    (tags_type, get_resource_name_completion_list, resource_group_name_type, enum_choice_list)
+
+from azure.cli.command_modules.datalake.store._validators import validate_resource_group_name
+from azure.mgmt.datalake.store.models.data_lake_store_account_management_client_enums \
+    import(FirewallState,
+           TrustedIdProviderState,
+           TierType,
+           FirewallAllowAzureIpsState)
+
+from azure.mgmt.datalake.store.models import (EncryptionConfigType)
+
+# ARGUMENT DEFINITIONS
+# pylint: disable=line-too-long
+datalake_store_name_type = CliArgumentType(help='Name of the Data Lake Store account.', options_list=('--account_name',), completer=get_resource_name_completion_list('Microsoft.DataLakeStore/accounts'), id_part=None)
+
+# PARAMETER REGISTRATIONS
+register_cli_argument('datalake store', 'resource_group_name', resource_group_name_type, id_part=None, required=False, help='If not specified, will attempt to discover the resource group for the specified Data Lake Store account.', validator=validate_resource_group_name)
+register_cli_argument('datalake store account show', 'name', datalake_store_name_type, options_list=('--account', '-n'))
+register_cli_argument('datalake store account delete', 'name', datalake_store_name_type, options_list=('--account', '-n'))
+register_cli_argument('datalake store', 'account_name', datalake_store_name_type, options_list=('--account', '-n'))
+register_cli_argument('datalake store account', 'tags', tags_type)
+register_cli_argument('datalake store account', 'tier', help='The desired commitment tier for this account to use.', **enum_choice_list(TierType))
+register_cli_argument('datalake store account create', 'resource_group_name', resource_group_name_type, completer=None, validator=None)
```

Comment:
Resource group name SHOULD have a completer here. Only the thing you are creating (the account) should not, since the whole point of creating a new account is that it doesn't exist.